core/src/avm2/namespace.rs

```diff
@@ -155,4 +232,49 @@ impl<'gc> Namespace<'gc> {
     pub fn as_uri(&self) -> AvmString<'gc> {
         self.as_uri_opt().unwrap_or_else(|| "".into())
     }
+
+    /// Compares two namespaces, requiring that their versions match exactly.
+    /// Normally, you should use `matches_ns,` which takes version compatibility
+    /// into account.
+    ///
+    /// Namespace does not implement `PartialEq`, so that each caller is required
+    /// to explicitly choose either `exact_version_match` or `matches_ns`.
+    pub fn exact_version_match(&self, other: Self) -> bool {
+        if Gc::as_ptr(self.0) == Gc::as_ptr(other.0) {
+            true
+        } else if self.is_private() || other.is_private() {
+            false
+        } else {
+            *self.0 == *other.0
+        }
+    }
+
+    /// Compares this namespace to another, considering them equal if this namespace's version
+    /// is less than or equal to the other (definitions in this namespace version can be
+    /// seen by the other). This is used to implement `ProperyMap`, where we want to
+    /// a definition with `ApiVersion::SWF_16` to be visible when queried from
+    /// a SWF with `ApiVersion::SWF_16` or any higher version.
+    pub fn matches_ns(&self, other: Self) -> bool {
+        if self.exact_version_match(other) {
+            return true;
+        }
+        match (&*self.0, &*other.0) {
+            (
+                NamespaceData::Namespace(name1, version1),
+                NamespaceData::Namespace(name2, version2),
+            ) => {
+                let name_matches = name1 == name2;
+                let version_matches = version1 <= version2;
+                if name_matches && !version_matches {
+                    tracing::info!(
```

Comment:
This `info!` feels like it's going to get triggered a lot- maybe an `avm_debug`?

Reply:
This should be rare - it only logs when namespace versioning is actually causing something to be hidden, which should be very rare in practice